zcash_client_sqlite/src/wallet/init/migrations/add_account_birthdays.rs

```diff
@@ -0,0 +1,81 @@
+//! This migration adds a birthday height to each account record.
+
+use std::collections::HashSet;
+
+use schemer_rusqlite::RusqliteMigration;
+use uuid::Uuid;
+use zcash_client_backend::data_api::{scanning::ScanPriority, SAPLING_SHARD_HEIGHT};
+
+use crate::wallet::{init::WalletMigrationError, scanning::priority_code};
+
+use super::v_sapling_shard_unscanned_ranges;
+
+pub(super) const MIGRATION_ID: Uuid = Uuid::from_u128(0xeeec0d0d_fee0_4231_8c68_5f3a7c7c2245);
+
+pub(super) struct Migration;
+
+impl schemer::Migration for Migration {
+    fn id(&self) -> Uuid {
+        MIGRATION_ID
+    }
+
+    fn dependencies(&self) -> HashSet<Uuid> {
+        [v_sapling_shard_unscanned_ranges::MIGRATION_ID]
+            .into_iter()
+            .collect()
+    }
+
+    fn description(&self) -> &'static str {
+        "Adds a birthday height for each account."
+    }
+}
+
+impl RusqliteMigration for Migration {
+    type Error = WalletMigrationError;
+
+    fn up(&self, transaction: &rusqlite::Transaction) -> Result<(), Self::Error> {
+        transaction.execute_batch(
+            "ALTER TABLE accounts ADD COLUMN birthday_height INTEGER;
+            UPDATE accounts SET birthday_height = MIN(blocks.height) FROM blocks;",
```

Comment:
I'm happy with this migration, because `init_blocks_table` would insert a row corresponding to the last height that wasn't scanned for the wallet (matching the intended semantics of `birthday_height`), and as of this migration being written we haven't had any production wallets inserting blocks in any other way.